tests/csapi/admin_test.go

```diff
@@ -13,3 +21,111 @@ func TestCanRegisterAdmin(t *testing.T) {
 	defer deployment.Destroy(t)
 	deployment.RegisterUser(t, "hs1", "admin", "adminpassword", true)
 }
+
+// Test if the implemented /_synapse/admin/v1/send_server_notice behaves as expected
+func TestServerNotices(t *testing.T) {
+	deployment := Deploy(t, b.BlueprintAlice)
+	defer deployment.Destroy(t)
+	admin := deployment.RegisterUser(t, "hs1", "admin", "adminpassword", true)
+	alice := deployment.Client(t, "hs1", "@alice:hs1")
+
+	reqBody := client.WithJSONBody(t, map[string]interface{}{
+		"user_id": "@alice:hs1",
+		"content": map[string]interface{}{
+			"msgtype": "m.text",
+			"body":    "hello from server notices!",
+		},
+	})
+	t.Run("/send_server_notice is not allowed as normal user", func(t *testing.T) {
+		res := alice.DoFunc(t, "POST", []string{"_synapse", "admin", "v1", "send_server_notice"})
+		must.MatchResponse(t, res, match.HTTPResponse{
+			StatusCode: http.StatusForbidden,
+			JSON: []match.JSON{
+				match.JSONKeyEqual("errcode", "M_FORBIDDEN"),
+			},
+		})
+	})
+	t.Run("/send_server_notice as an admin is allowed", func(t *testing.T) {
+		sendServerNotice(t, admin, reqBody, nil)
+
+	})
+	t.Run("Alice is invited to the server alert room", func(t *testing.T) {
+		sendServerNotice(t, admin, reqBody, nil)
+		syncUntilInvite(t, alice)
+	})
+	t.Run("Alice cannot reject the invite", func(t *testing.T) {
+		sendServerNotice(t, admin, reqBody, nil)
```

Comment:
Why are we constantly sending server notices? These subtests will run sequentially.

Reply:
I was running them in parallel first, but remembered that this might race since there can only be one active notice room. Fixed.